web_interface/blueprints/api_v3.py

```
@@ -1384,6 +1384,52 @@ def get_system_version():
+@api_v3.route('/system/check-update', methods=['GET'])
+def check_for_update():
+    """Check whether a newer LEDMatrix commit is available on origin/main."""
+    now = time.time()
+    if _update_check_cache['result'] and now - _update_check_cache['ts'] < _UPDATE_CHECK_TTL:
+        return jsonify(_update_check_cache['result'])
+
+    _safe: Dict[str, Any] = {'update_available': False, 'remote_sha': 'unknown', 'commits_behind': 0}
+    try:
+        cwd = str(PROJECT_ROOT)
+        subprocess.run(
+            ['git', 'fetch', 'origin', 'main', '--quiet'],
```

Ruff (0.15.13)

[error] Starting a process with a partial executable path

(S607)